apps/tasks/messages.js

```diff
@@ -72,10 +76,8 @@ module.exports = function (grunt) {
       }
     }
 
-    return [
-      'var ' + namespace + ' = ' + mf.functions() + ';',
```

Comment:
can you explain the changes here? It looks at a glance like the new version is missing a lot compared to the old version; I assume that's just because the new compile process takes care of some things for us, but I'd love some more informed insights.

Reply:
https://messageformat.github.io/messageformat/MessageFormat#compile__anchor has a bit of documentation. My understanding is that now the compile method produces JS that's intended to be written to a file and imported/used later (you can see that in the third example in the documentation). From that documentation: "If the input messages -- and therefore the output -- of compile() is an object, the output object will have a toString(global) method that may be used to store or cache the compiled functions to disk, for later inclusion in any JS environment, without a local MessageFormat instance required." In our case we don't want to create a local MessageFormat instance to use these strings but just want to assign them to `window.blockly`.

The first part of one of the build files now looks like:
```
var es_mx = function (n, ord
) {
  if (ord) return 'other';
  return (n == 1) ? 'one' : 'other';
};
var number = function (value, name, offset) {
  if (!offset) return value;
  if (isNaN(value)) throw new Error("Can't apply offset:" + offset + ' to argument `' + name + '` with non-numerical value ' + JSON.stringify(value) + '.');
  return value - offset;
};
var plural = function (value, offset, lcfunc, data, isOrdinal) {
  if ({}.hasOwnProperty.call(data, value)) return data[value];
  if (offset) value -= offset;
  var key = lcfunc(value, isOrdinal);
  return key in data ? data[key] : data.other;
};

(window.blockly = window.blockly || {}).common_locale = {
  abuseDetail: function(d) { return "Proporcione detalles sobre el contenido que está informando"; },
...
```

Those functions that we manually added earlier are now added by default it seems.

Reply:
Perfect! That's exactly what I was hoping the explanation would be